src/lib/SaferNames/Name.hs

```diff
@@ -91,7 +94,7 @@ class Monad1 m => ScopeReader (m::MonadKind1) where
   getScope :: m n (WithDistinct Scope n)
 
 class ScopeReader m => ScopeExtender (m::MonadKind1) where
-  extendScope :: Distinct l => ScopeFrag n l -> m l r -> m n r
+  extendScope :: Distinct l => ScopeFrag n l -> (Ext n l => m l r) -> m n r
```

Comment:
What does `Ext n l` mean?

Reply:
It means that `l` is a (not-necessarily-fresh) extension of `n`. See 8ffde6164c07c020b3831d620b12a1dcf157dce6

Reply:
Is that not implied by the existence of `ScopeFrag n l`? How can you have a scope fragment when the target is not an extension?

Reply:
Ah I guess that this `Ext` is in a negative position, so you're only adding more evidence to the callback. Fair enough